playbooks/roles/genesis-exoscale/tasks/main.yml

```diff
@@ -0,0 +1,178 @@
+---
+- name: Set the Exoscale API details facts to the value that was entered, or attempt to retrieve it from the environment if the entry is blank
+  set_fact:
+    EXO_API_SECRET: "{{ EXO_API_SECRET_entry | default( lookup('env', 'EXO_API_SECRET') ) }}"
+    EXO_API_KEY: "{{ EXO_API_KEY_entry | default( lookup('env', 'EXO_API_KEY') ) }}"
+    EXO_API_ENDPOINT: "{{ lookup('env', 'EXO_API_ENDPOINT') | default('https://api.exoscale.ch/compute', true) }}"
+
+
+- name: Get the default SSH key
+  command: cat ~/.ssh/id_rsa.pub
+  register: ssh_key
+
+- block:
+    - name: Add the SSH key to Exoscale if it doesn't already exist
+      local_action:
+        module: cs_sshkeypair
+        name: "{{ exo_ssh_name }}"
+        public_key: "{{ ssh_key.stdout }}"
+      register: exo_ssh_key
+  rescue:
+    - fail:
+        msg: "* The API key or secret might be incorrect or missing the Write Scope. OR * The SSH key may already exist in the Exoscale portal under a different name. OR * The cs Python module might not be installed. Use `pip install cs` to install this module."
+
+- block:
+  - name: Streisand Security Group
+    local_action:
+      module: cs_securitygroup
+      name: "streisand"
```

Comment:
Streisand creates unique GCE networks and AWS security groups for each server by incorporating the server name into the identifier. We likely want to do the same thing on Exoscale to prevent any conflicts if users are running multiple Streisand servers within their account. This will also be helpful in the future when different servers could be running different combinations of the various VPN daemons and a one-size-fits-all security group will no longer be ideal.